### PR TITLE
feat(ci): website deployment IaC and github actions

### PR DESCRIPTION
**Pull request by zackpollard**
Deploy website using github actions instead of letting cloudflare do it

Added terraform for managing DNS records so we have all previews etc going through our main domain

### CODE REVIEW

File: .github/workflows/docs-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      - name: Publish to Cloudflare Pages
+        uses: cloudflare/pages-action@v1
+        with:
+          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN_PAGES_UPLOAD }}
+          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
+          projectName: "immich-app"
+          workingDirectory: "docs"
+          directory: "build"
+          branch: ${{ steps.parameters.outputs.name }}
+          wranglerVersion: '3'
```
Comment (bo0tzz) [inline .github/workflows/docs-deploy.yml:148]: Should we publish before we update the domains, so we don't end up in a situation where the publish fails and we're left with the main docs pointing at an empty deployment?

File: .github/workflows/docs-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+      - name: Comment
+        uses: actions-cool/maintain-one-comment@v3
+        if: ${{ steps.parameters.outputs.event == 'pr' }}
+        with:
+          number: ${{ github.event.workflow_run.pull_requests[0].number }}
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (bo0tzz) [inline .github/workflows/docs-deploy.yml:155]: We shouldn't need to set this explicitly, the action picks it up by default.

**Review by bo0tzz — COMMENTED** [review]
Cool!
Since I don't know terraform I'll just have to trust you that it all works ;)

File: .github/workflows/docs-deploy.yml (hunk 1 above)
Comment (zackpollard) [reply .github/workflows/docs-deploy.yml:148]: Yea good shout

**Review by bo0tzz — APPROVED** [review]

**Pull request merged**